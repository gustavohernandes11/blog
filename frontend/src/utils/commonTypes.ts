

interface ImageAttributes {
    url: string;
    height: number;
    width: number;
}

interface CapeData {
    attributes: ImageAttributes;
}

interface Cape {
    data: CapeData;
}

interface Meta {
    Description: string;
    Title: string;
    Keywords: string;
}

interface PostAttributes {
    Cape: Cape;
    Meta: Meta;
    Title: string;
    Content: string;
    Reference: string;
    Excerpt: string;
    Slug: string;
    Category: string;
    publishedAt: string
}

export interface Post {
    id: string;
    attributes: PostAttributes;
}

export interface Posts {
    map(arg0: (e: Post) => { params: { slug: string; }; }): unknown;
    data: Post[];
}

export interface Data {
    posts: Posts;
}

export interface GraphqlResponse {
    data: Data;
}

export type loadPostsVariables = {
    titleContains?: string;
    category?: string;
    postSlug?: string;
    postSearch?: string;
    sort?: string;
    limit?: number;
    start?: number;
};
