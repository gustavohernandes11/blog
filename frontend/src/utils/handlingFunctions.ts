export const getDate = (inputDate: string) => {
    const date = new Date(inputDate);
    const meses = [
        "Jan",
        "Fev",
        "Mar",
        "Abr",
        "Mai",
        "Jun",
        "Jul",
        "Ago",
        "Set",
        "Out",
        "Nov",
        "Dez",
    ];
    const formatedDate =
        date.getDate() +
        " / " +
        meses[date.getMonth()] +
        " / " +
        date.getFullYear();
    return formatedDate;
};
