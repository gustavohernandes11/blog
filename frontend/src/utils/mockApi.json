{
    "data": [
      {
        "id": 1,
        "attributes": {
          "Post_description": "Lorem ?",
          "createdAt": "2022-07-25T21:06:00.456Z",
          "updatedAt": "2022-07-25T22:25:36.066Z",
          "publishedAt": "2022-07-25T21:06:09.397Z",
          "locale": "en",
          "Title": "Lorem ipsum ",
          "Post_content_zone": [
            {
              "id": 1,
              "__component": "post.quote",
              "quote": "\n>Lorem\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n"
            },
            {
              "id": 1,
              "__component": "post.paragraphy",
              "paragraphy": "lorem lorem?\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n"
            }
          ],
          "Meta": {
            "id": 1
          },
          "localizations": {
            "data": []
          }
        }
      },
      {
        "id": 2,
        "attributes": {
          "Post_description": "\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi \n\n",
          "createdAt": "2022-07-25T21:30:31.032Z",
          "updatedAt": "2022-07-25T22:35:59.952Z",
          "publishedAt": "2022-07-25T22:35:59.464Z",
          "locale": "en",
          "Title": "Lorem i\"Lorem laborum. ",
          "Post_content_zone": [
            {
              "id": 2,
              "__component": "post.paragraphy",
              "paragraphy": "\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n"
            },
            {
              "id": 2,
              "__component": "post.quote",
              "quote": "\n>\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n\n"
            },
            {
              "id": 3,
              "__component": "post.paragraphy",
              "paragraphy": "\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\n"
            }
          ],
          "Meta": null,
          "localizations": {
            "data": []
          }
        }
      },
      {
        "id": 3,
        "attributes": {
          "Post_description": "Lorem lorem lorem lorem ipsum",
          "createdAt": "2022-07-25T22:27:00.444Z",
          "updatedAt": "2022-07-25T22:49:14.115Z",
          "publishedAt": "2022-07-25T22:49:13.400Z",
          "locale": "en",
          "Title": "Lorem Lorem ",
          "Post_content_zone": [
            {
              "id": 4,
              "__component": "post.paragraphy",
              "paragraphy": "\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n"
            },
            {
              "id": 5,
              "__component": "post.paragraphy",
              "paragraphy": "\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n\"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.\"\n"
            }
          ],
          "Meta": {
            "id": 2
          },
          "localizations": {
            "data": []
          }
        }
      }
    ],
    "meta": {
      "pagination": {
        "page": 1,
        "pageSize": 25,
        "pageCount": 1,
        "total": 3
      }
    }
  }