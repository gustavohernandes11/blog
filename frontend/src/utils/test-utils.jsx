import { render } from "@testing-library/react";
import { Theme } from "../theme/ThemeProvider";

export const customRender = (ui, options) =>
    render(ui, { wrapper: Theme, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
