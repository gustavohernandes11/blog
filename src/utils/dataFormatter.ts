export const dateFormatter = (raw: string) => Intl.DateTimeFormat("pt-BR", {
    dateStyle: "long",
}).format(new Date(raw));
