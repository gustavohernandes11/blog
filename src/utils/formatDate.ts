/**
 * Returns the formatted data from ISO to Brazilian locale string.
 *
 * @example
 * // returns "6 de janeiro de 2024"
 * formatDate("2024-01-06T21:40:17.893Z");
 *
 * @param isoString - The data string in ISO format.
 * @returns A friendly version of the date.
 */
export const formatDate = (isoString: string): string => {
    return new Date(isoString).toLocaleString("pt-BR", {
        year: "numeric",
        month: "long",
        day: "numeric",
    });
};
