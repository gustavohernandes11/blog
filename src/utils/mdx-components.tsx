import SyntaxHighlighter from "react-syntax-highlighter";
import { atomDark } from "react-syntax-highlighter/dist/cjs/styles/prism";
import { ArticleHeading } from "../components/ArticleHeading";

export const MDXComponents = {
    h1: ({ children, ...props }: any) => (
        <ArticleHeading as="h1" {...props}>
            {children}
        </ArticleHeading>
    ),
    h2: ({ children, ...props }: any) => (
        <ArticleHeading as="h2" {...props}>
            {children}
        </ArticleHeading>
    ),
    h3: ({ children, ...props }: any) => (
        <ArticleHeading as="h3" {...props}>
            {children}
        </ArticleHeading>
    ),
    h4: ({ children, ...props }: any) => (
        <ArticleHeading as="h4" {...props}>
            {children}
        </ArticleHeading>
    ),
    h5: ({ children, ...props }: any) => (
        <ArticleHeading as="h5" {...props}>
            {children}
        </ArticleHeading>
    ),
    h6: ({ children, ...props }: any) => (
        <ArticleHeading as="h6" {...props}>
            {children}
        </ArticleHeading>
    ),
    code({ node, inline, className, style, children, ...props }: any) {
        const match = /language-(\w+)/.exec(className || "");
        return !inline && match ? (
            <SyntaxHighlighter
                children={String(children).replace(/\n$/, "")}
                showLineNumbers
                lineNumberStyle={{ minWidth: 0 }}
                style={atomDark}
                language={match[1]}
                {...props}
            />
        ) : (
            <code className={className} {...props}>
                {children}
            </code>
        );
    },
};
