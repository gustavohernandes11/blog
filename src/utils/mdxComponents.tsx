import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { vscDarkPlus } from "react-syntax-highlighter/dist/cjs/styles/prism";
import { ArticleHeading } from "../components/ArticleHeading";
import { Flex } from "components/Flex";

export const MDXComponents = {
    img: (props: any) => (
        <Flex justify="center">
            <img {...props} />
        </Flex>
    ),
    h1: ({ children, ...props }: any) => (
        <ArticleHeading as="h1" {...props}>
            {children}
        </ArticleHeading>
    ),
    h2: ({ children, ...props }: any) => (
        <ArticleHeading as="h2" {...props}>
            {children}
        </ArticleHeading>
    ),
    h3: ({ children, ...props }: any) => (
        <ArticleHeading as="h3" {...props}>
            {children}
        </ArticleHeading>
    ),
    h4: ({ children, ...props }: any) => (
        <ArticleHeading as="h4" {...props}>
            {children}
        </ArticleHeading>
    ),
    h5: ({ children, ...props }: any) => (
        <ArticleHeading as="h5" {...props}>
            {children}
        </ArticleHeading>
    ),
    h6: ({ children, ...props }: any) => (
        <ArticleHeading as="h6" {...props}>
            {children}
        </ArticleHeading>
    ),
    code(props: any) {
        const { children, className, node, ...rest } = props;
        const match = /language-(\w+)/.exec(className || "");
        return match ? (
            <SyntaxHighlighter
                {...rest}
                children={String(children).replace(/\n$/, "")}
                style={vscDarkPlus}
                language={match[1]}
                PreTag="div"
            />
        ) : (
            <code {...rest} className={className}>
                {children}
            </code>
        );
    },
};
