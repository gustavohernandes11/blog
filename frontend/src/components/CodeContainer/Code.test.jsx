import { screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/jest-dom";

import { customRender } from "../../utils/test-utils";
import { CodeContainer } from "./index";

describe("<CodeContainer />", () => {
    it("should render the component <CodeContainer />", () => {
        const { container } = customRender(
            <CodeContainer>Texto para teste</CodeContainer>
        );

        expect(container).toMatchSnapshot();
    });
    it("should render the content in the screen", () => {
        customRender(<CodeContainer>Texto para teste</CodeContainer>);

        const text = screen.getByText(/texto para teste/i);
        expect(text).toBeInTheDocument();
    });
    describe('<IconButton aria-label="copy-button" />', () => {
        let codeContainer = null;
        beforeAll(() => {
            customRender(<CodeContainer>Texto para teste</CodeContainer>);
            codeContainer = screen.getByText(/texto para teste/i);
        });
        it("should render the copy-button on mouseover event", () => {
            fireEvent.mouseOver(codeContainer);
            const copyButton = screen.getByLabelText("copy-code-button");
            expect(copyButton).toBeInTheDocument();
        });
        it("should not render the copy-button on mouseleave event", () => {
            fireEvent.mouseOver(codeContainer);
            const copyButton = screen.queryByAltText("copy-code-button");
            fireEvent.mouseLeave(codeContainer);
            expect(copyButton).not.toBeInTheDocument();
        });
    });
});
