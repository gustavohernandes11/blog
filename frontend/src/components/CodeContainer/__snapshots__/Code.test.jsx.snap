// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeContainer /> should render the component <CodeContainer /> 1`] = `
<div>
  <code
    aria-label="code-container"
    class="styles__Container-sc-1qlmxzr-0 cJSWVw"
  >
    Texto para teste
  </code>
</div>
`;
